src/wallet/api/wallet.cpp

```diff
@@ -656,9 +656,11 @@ string WalletImpl::keysFilename() const
     return m_wallet->get_keys_file();
 }
 
-bool WalletImpl::init(const std::string &daemon_address, uint64_t upper_transaction_size_limit)
+bool WalletImpl::init(const std::string &daemon_address, uint64_t upper_transaction_size_limit, const std::string &daemon_username, const std::string &daemon_password)
```

Comment:
I would take by value, then `std::move` each string. It reduces a copy if the caller provides a temporary or moves the string when calling `WalletImpl::init`.